vermouth/processors/apply_rubber_band.py

```diff
@@ -158,30 +163,24 @@ def build_connectivity_matrix(graph, separation, selected_nodes):
     numpy.ndarray
         A boolean matrix.
     """
-    if separation < 0:
-        raise ValueError('Separation has to be zero or positive.')
-    if separation == 0:
-        # The connectivity matrix with a separation of 0 is the adjacency
-        # matrix. Thankfully, networkx can directly give it to us a a numpy
-        # matrix.
-        return np.asarray(nx.to_numpy_matrix(graph, nodelist=selected_nodes).astype(bool))
+    res_graph = make_residue_graph(graph)
+    distance_pairs = nx.all_pairs_shortest_path_length(res_graph, cutoff=separation)
+    # only gets "positive" entries due to the cutoff argument above
     size = graph.number_of_nodes()
-    correspondence = {node: index for index, node in enumerate(graph.nodes)}
     # the matrix will be reduced before returning it but nx.all_pairs_shortest_path_length
     # does not take a subset of nodes
     connectivity = np.zeros((size, size), dtype=bool)
```

Comment:
TODO: optimize me, try to create a sparse matrix in case scipy is available.
Please add a comment